### ISSUE TITLE
How to compute loss in training?

### ISSUE THREAD

**Issue by dtdo90** [lyuwenyu/RT-DETR#511]
Dear author,

Thank you for the amazing work! I have a question on computing loss.

The loss output contains 3 main losses 'loss_vfl', 'loss_bbox', 'loss_giou' and auxiliary losses for vfl, bbox, giou. The criterion is 

'RTDETRCriterionv2': {
	'weight_dict': {'loss_vfl': 1, 'loss_bbox': 5, 'loss_giou': 2}, 
	'losses': ['vfl', 'boxes'], 
	'alpha': 0.75, 'gamma': 2.0, 
	'matcher': {'type': 'HungarianMatcher', 'weight_dict': {'cost_class': 2, 'cost_bbox': 5, 'cost_giou': 2}, 'alpha': 0.25, 'gamma': 2.0}}

What is the formula for computing the total loss for training (in loss.backwards())? Is it  1*loss_vfl+ 5*loss_bbox+2*loss_giou? How about the auxiliary losses?

I very much look forward to your answer!

**Comment by dtdo90** [comment]
It seems 'weight_dict' is already integrated in the output loss dictionary. So would the total loss be the sum of all loss components including the auxiliary losses?

**Comment by lyuwenyu** [comment]
> What is the formula for computing the total loss for training (in loss.backwards())? Is it 1_loss_vfl+ 5_loss_bbox+2*loss_giou? How about the auxiliary losses?

Yes, It is same for each decoder layer. 

---


> So would the total loss be the sum of all loss components including the auxiliary losses?

Yes.

**Comment by dtdo90** [comment]
Thank you for the quick reply and confirmation!
